shared/constants/types/flux.js

```diff
@@ -19,6 +19,6 @@ export type NoErrorTypedAction<T, P> = TypedAction<T, P, P>
 export type Action = TypedAction<any, any, any>
 export type GetState = () => any
 export type AsyncAction = (dispatch: Dispatch, getState: GetState) => ?Promise<*> // eslint-disable-line no-use-before-define
-export type Dispatch = (action: AsyncAction | Action) => ?Promise<*> // eslint-disable-line no-use-before-define
```

Comment:
our dispatch actions shouldn't ever return promises in practice. that should be captured in redux somewhere instead

Reply:
also a nice comment